pkg/apis/pipeline/v1alpha2/param_types.go

```diff
@@ -124,3 +125,77 @@ func (arrayOrString *ArrayOrString) ApplyReplacements(stringReplacements map[str
 		arrayOrString.ArrayVal = newArrayVal
 	}
 }
+
+// ResultReference is a type that represents a reference to a task run result
+type ResultReference struct {
+	PipelineTask string
+	Result       string
+}
+
+const (
+	resultExpressionFormat = "tasks.<taskName>.results.<resultName>"
+	resultTaskPart         = "tasks"
+	resultResultPart       = "results"
+)
+
+// NewResultReference extracts a ResultReference form param.
+// If the ResultReference cab be extracted, it is returned. Otherwise an error is returned
+func NewResultReference(param Param) (*ResultReference, error) {
+	substitutionExpression, ok := getVarSubstitutionExpression(param)
+	if !ok {
+		return nil, fmt.Errorf("Invalid result reference expression: must contain variable substitution %q", resultExpressionFormat)
+	}
+	pipelineTask, result, err := parseExpression(substitutionExpression)
+	if err != nil {
+		return nil, fmt.Errorf("Invalid result reference expression: %v", err)
+	}
+	return &ResultReference{
+		PipelineTask: pipelineTask,
+		Result:       result,
+	}, nil
+}
+
+// HasResultReference is similar to NewResultReference, but it is used
+// when the error is not relevant. i.e. we simply want to check if param contains
+// a result reference
+func HasResultReference(param Param) (*ResultReference, bool) {
```

Comment:
While I understand the impulse to add sugar for common use cases I'd argue that this doesn't add a ton of value over simply doing something like this directly in the 2 places which use `HasResultReference`:

```go
if resultRef, _ := NewResultReference(param); resultRef != nil {
  // The param is a result reference
}
```